### PR TITLE
Reword missing API error to mention client update

### PR DESCRIPTION
**Pull request by dsimansk**
## Description

Add a mention of client update to common missing API error message. In case of older client with newer backend with removed API, it might confusing why there's a failure.

## Changes

* Reword missing API error to mention client update

## Reference

Fixes #

### CODE REVIEW

File: pkg/errors/errors.go [hunk 1]
```diff
@@ -17,12 +17,15 @@ package errors
...
+	"unicode/utf8"
 )
 
-func newInvalidCRD(apiGroup string) *KNError {
+func NewInvalidCRD(apiGroup string) *KNError {
 	parts := strings.Split(apiGroup, ".")
 	name := parts[0]
-	msg := fmt.Sprintf("no Knative %s API found on the backend, please verify the installation", name)
+	msg := fmt.Sprintf("no Knative %s API found on the backend, please verify the installation or "+
+		"update the 'kn' client to latest version", firstCharToUpper(name))
```
Comment (dsimansk) [inline pkg/errors/errors.go:28]: Might be better to call it `matching version`?

Comment (rhuss) [reply pkg/errors/errors.go:28]: I wonder whether 'no Knative API' found is appropriate as there could be a Knative API available but too new. So maybe rephrase to "no or newer Knative %s API found on the backend, please verify the installation or update this client"

I just would say update, as it's kind of clear that it should be moved to a matching client.

File: pkg/errors/errors.go [hunk 2]
```diff
@@ -37,3 +40,11 @@ func newNoRouteToHost(errString string) *KNError {
 func newNoKubeConfig(errString string) *KNError {
 	return NewKNError("no kubeconfig has been provided, please use a valid configuration to connect to the cluster")
 }
+
+func firstCharToUpper(s string) string {
```
Comment (rhuss) [inline pkg/errors/errors.go:44]: +1

Actually, I would use this function, too, before emitting an error message to the console, so that error messages that start with a lowercase letter (golang convention) will be upcased. There should be a single exit-point for errors to do the massaging of the error message to print. But that's of course another story ;-)

**Review by rhuss — COMMENTED** [review]
Loogs good, one minor suggestion wrt/ wording.

File: pkg/errors/errors.go (hunk 1 above)
Comment (dsimansk) [reply pkg/errors/errors.go]: @rhuss addressed in latest update.

**Review by rhuss — COMMENTED** [review]
/approve
/lgtm
